File: patch.py
import re

with open(r'c:\Users\user\sat_project_12\study-planner.html', 'r', encoding='utf-8') as f:
    content = f.read()

# 1. JS: Update mockupTasks with rowBg
content = content.replace(
    'border: "#378ADD", badgeBg: "#DBEAFE"',
    'border: "#378ADD", badgeBg: "#DBEAFE", rowBg: "#E6F1FB33"'
)
content = content.replace(
    'border: "#7F77DD", badgeBg: "#FAE8FF"',
    'border: "#7F77DD", badgeBg: "#FAE8FF", rowBg: "#EEEDFE33"'
)
content = content.replace(
    'border: "#1D9E75", badgeBg: "#FEF3C7"',
    'border: "#1D9E75", badgeBg: "#FEF3C7", rowBg: "#E1F5EE33"'
)

# 2. JS: Remove default seed for geometry
seed_block = """if (checkedTasks['mock_geometry'] === undefined) {
        checkedTasks['mock_geometry'] = true;
        checkedTasks['mock_heart_of_algebra'] = false;
        checkedTasks['mock_command_of_evidence'] = false;
        checkedTasks['mock_rhetorical_synthesis'] = false;
        checkedTasks['mock_daily_vocabulary'] = false;
        localStorage.setItem('oneprep_plan_checks', JSON.stringify(checkedTasks));
      }"""
content = content.replace(seed_block, '')

# 3. JS: Remove truncation from h4, add border-radius, use rowBg
content = content.replace(
    '''border-radius: 0 var(--lav-radius, 20px) var(--lav-radius, 20px) 0; border-left: 3px solid ${task.border};''',
    '''border-radius: 0 8px 8px 0; border-left: 3px solid ${task.border};'''
)
content = content.replace(
    '''const cardBg = isChecked ? '#F8F9FA' : '#ffffff';''',
    '''const cardBg = task.rowBg;'''
)
content = content.replace(
    '''h4 style="margin: 0; font-size: 13px; font-weight: 500; color: ${isChecked ? 'var(--color-text-secondary, #A0A0B0)' : 'var(--lav-dark)'}; line-height: 1.4; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; max-width: 170px;''',
    '''h4 style="margin: 0; font-size: 13px; font-weight: 500; color: ${isChecked ? 'var(--color-text-secondary, #A0A0B0)' : 'var(--lav-dark)'}; line-height: 1.4;'''
)

# 4. HTML: Table headers Ahead -> Upcoming
content = content.replace('&mdash; Ahead', '&mdash; Upcoming')

# 5. HTML: Table rows left border and tints
def patch_td(match, border_color, bg_color):
    td = match.group(0)
    # remove overflow, text-overflow, white-space
    td = re.sub(r'overflow:\s*hidden;\s*text-overflow:\s*ellipsis;\s*white-space:\s*nowrap;', '', td)
    # set border-left
    td = re.sub(r'border-left:\s*[^;]+;', f'border-left: 3px solid {border_color};', td)
    # add/replace background inside td
    if 'background:' in td:
        td = re.sub(r'background:\s*#[0-9a-fA-F]+;?', f'background: {bg_color};', td)
    else:
        td = re.sub(r'(style="[^"]*)(")', rf'\1 background: {bg_color};\2', td)
    # remove max-width
    td = re.sub(r'max-width:\s*[^;]+;', '', td)
    return td

def process_tr(match):
    tr = match.group(0)
    if 'Heart of Algebra' in tr or 'Geometry' in tr or 'Mock Exam' in tr or 'Advanced Math' in tr or 'Full Math' in tr:
        border = '#378ADD'
        bg = '#E6F1FB22'
    elif 'Command of Evidence' in tr or 'Rhetorical Synthesis' in tr or 'Reading Comp' in tr or 'Full R&W' in tr:
        border = '#7F77DD'
        bg = '#EEEDFE22'
    elif 'Daily Vocabulary' in tr:
        border = '#1D9E75'
        bg = '#E1F5EE22'
    else:
        border = '#000000'
        bg = 'transparent'
    
    tr = re.sub(r'<td\s+style="[^"]+">', lambda m: patch_td(m, border, bg), tr)
    tr = re.sub(r'<div\s+style="overflow:\s*hidden;\s*text-overflow:\s*ellipsis;\s*white-space:\s*nowrap;">', '<div>', tr)
    return tr

tbody_start = content.find('<tbody>')
tbody_end = content.find('</tbody>', tbody_start)
tbody_content = content[tbody_start:tbody_end]
patched_tbody = re.sub(r'(?s)<tr>.*?</tr>', process_tr, tbody_content)
content = content[:tbody_start] + patched_tbody + content[tbody_end:]

# 6. HTML: Heights
content = content.replace(
    '<aside class="sp-tasks-col" style="flex: 0 0 42%; display: flex; flex-direction: column; min-height: 0;">',
    '<aside class="sp-tasks-col" style="flex: 0 0 42%; display: flex; flex-direction: column; height: 100%;">'
)
content = content.replace(
    '<div class="sp-table-card" style="display: flex; flex-direction: column; flex: 1; padding: 0; padding-bottom: 24px;">',
    '<div class="sp-table-card" style="display: flex; flex-direction: column; height: 100%; flex: 1; padding: 0; padding-bottom: 24px;">'
)

# 7. HTML: Update static tasks
def patch_static_tasks(match):
    div = match.group(0)
    if 'border-left-color:#3B82F6;' in div:
        border = '#378ADD'
        bg = '#E6F1FB33'
    elif 'border-left-color:#7C6FE0;' in div:
        border = '#7F77DD'
        bg = '#EEEDFE33'
    elif 'border-left-color:#10B981;' in div:
        border = '#1D9E75'
        bg = '#E1F5EE33'
    else:
        border = '#000000'
        bg = 'transparent'
    
    # ensure border-radius is 0 8px 8px 0
    div = re.sub(r'style="([^"]*)"', lambda m: f'style="{m.group(1)} border-left: 3px solid {border}; background: {bg}; border-radius: 0 8px 8px 0;"', div, count=1)
    return div

dac_tasks_start = content.find('<div id="daily-agenda-tasks"')
dac_tasks_end = content.find('</aside>', dac_tasks_start)
dac_content = content[dac_tasks_start:dac_tasks_end]
patched_dac = re.sub(r'<div class="da-task-card.*?</div>\s*</div>', patch_static_tasks, dac_content, flags=re.DOTALL)
content = content[:dac_tasks_start] + patched_dac + content[dac_tasks_end:]

with open(r'c:\Users\user\sat_project_12\study-planner.html', 'w', encoding='utf-8') as f:
    f.write(content)
print("Patched successfully.")

File: patch.js
const fs = require('fs');

const filePath = 'c:\\Users\\user\\sat_project_12\\study-planner.html';
let content = fs.readFileSync(filePath, 'utf-8');

// 1. JS mockupTasks Backgrounds (Today's Agenda)
// Update the rowBg in the JS array to use the new 44 opacity for agenda
content = content.replace(/rowBg: "#E6F1FB33"/g, 'rowBg: "#E6F1FB44"');
content = content.replace(/rowBg: "#EEEDFE33"/g, 'rowBg: "#EEEDFE44"');
content = content.replace(/rowBg: "#E1F5EE33"/g, 'rowBg: "#E1F5EE44"');

// 2. JS Template String
// Replace border-left: 3px solid ${task.border} -> border-left: 4px solid ${task.border} !important
// background: ${cardBg} -> background: ${cardBg} !important
// border-radius: 0 8px 8px 0 -> border-radius: 0 8px 8px 0 !important
// add padding-left: 12px !important
content = content.replace(
    /background: \$\{cardBg\};.*?border-radius: 0 8px 8px 0;\s*border-left: 3px solid \$\{task\.border\};/,
    'background: ${cardBg} !important; opacity: ${cardOp}; border: 0.5px solid var(--color-border-tertiary, var(--lav-border)); border-radius: 0 8px 8px 0 !important; border-left: 4px solid ${task.border} !important; padding-left: 12px !important;'
);

// 3. Static Today's Agenda (HTML)
const dacStart = content.indexOf('<div id="daily-agenda-tasks"');
const dacEnd = content.indexOf('</aside>', dacStart);
let dacContent = content.substring(dacStart, dacEnd);

dacContent = dacContent.replace(/<div class="da-task-card.*?<\/div>\s*<\/div>/gs, (div) => {
    // we want to replace the style block with the new !important rules
    return div.replace(/style="([^"]*)"/, (match, p1) => {
        // Remove existing border-left, background, border-radius, padding-left if any
        let newStyle = p1.replace(/border-left:\s*3px\s*solid\s*#[0-9a-fA-F]+;/i, '');
        newStyle = newStyle.replace(/background:\s*#[0-9a-fA-F]+;/i, '');
        newStyle = newStyle.replace(/border-radius:\s*0\s*8px\s*8px\s*0;/i, '');
        
        let border = '#000000';
        let bg = 'transparent';
        if (div.includes('border-left-color:#3B82F6;') || div.includes('Math')) {
            border = '#378ADD';
            bg = '#E6F1FB44';
        } else if (div.includes('border-left-color:#7C6FE0;') || div.includes('English')) {
            border = '#7F77DD';
            bg = '#EEEDFE44';
        } else if (div.includes('border-left-color:#10B981;') || div.includes('Vocab')) {
            border = '#1D9E75';
            bg = '#E1F5EE44';
        }
        
        return `style="${newStyle.trim()} border-left: 4px solid ${border} !important; background: ${bg} !important; border-radius: 0 8px 8px 0 !important; padding-left: 12px !important;"`;
    });
});

content = content.substring(0, dacStart) + dacContent + content.substring(dacEnd);

// 4. Weekly Table (HTML)
const patchTd = (td, border, bg) => {
    let result = td.replace(/border-left:\s*3px\s*solid\s*#[0-9a-fA-F]+;/, `border-left: 4px solid ${border} !important;`);
    result = result.replace(/background:\s*#[0-9a-fA-F]+;/, `background: ${bg} !important;`);
    return result;
};

const tbodyStart = content.indexOf('<tbody>');
const tbodyEnd = content.indexOf('</tbody>', tbodyStart);
let tbodyContent = content.substring(tbodyStart, tbodyEnd);

tbodyContent = tbodyContent.replace(/<tr>.*?<\/tr>/gs, (tr) => {
    let border = '#000000';
    let bg = 'transparent';
    if (/Heart of Algebra|Geometry|Mock Exam|Advanced Math|Full Math/.test(tr)) {
        border = '#378ADD';
        bg = '#E6F1FB33';
    } else if (/Command of Evidence|Rhetorical Synthesis|Reading Comp|Full R&W/.test(tr)) {
        border = '#7F77DD';
        bg = '#EEEDFE33';
    } else if (/Daily Vocabulary/.test(tr)) {
        border = '#1D9E75';
        bg = '#E1F5EE33';
    }

    let patchedTr = tr.replace(/<td\s+style="[^"]+">/g, (td) => patchTd(td, border, bg));
    return patchedTr;
});

content = content.substring(0, tbodyStart) + tbodyContent + content.substring(tbodyEnd);

fs.writeFileSync(filePath, content, 'utf-8');
console.log("Patched successfully.");
